### PR TITLE
Helper functions `tryTake`, `tryTakeCopy` to lookup in key-value tables

### PR DESCRIPTION
**Pull request by akindyakov**
There are a lot of lookups in the maps the osquery code. Most of them are verbose and not-optimal with check if such key exists in the table before get access. Some of them consists error e.g.:
```c++
    r["uid"] = row.count("uuid") > 0 ? row.at("uid") : "";
```
Introduced code will help to avoid the most of such problems.

### CODE REVIEW

File: osquery/core/tests/get_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
```
Comment (fmanco) [inline osquery/core/tests/get_tests.cpp]: Header as above

File: osquery/core/tests/get_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      {"key", "value"},
+      {"kKey", "vValue"},
+  };
+  ASSERT_EQ(m.size(), 2);
+  {
+    auto const exp = tryTakeCopy(m, "key");
+    ASSERT_TRUE(exp.isValue());
+    EXPECT_EQ(exp.get(), "value");
+  }
+  {
```
Comment (fmanco) [inline osquery/core/tests/get_tests.cpp]: Maybe assert container size didn't change.

File: osquery/core/tests/get_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  TestValue& operator=(TestValue&&) = delete;
+
+ public:
+  std::string text;
+};
+
+template <typename MapType>
+void testGetOr() {
+  auto m = MapType{};
+  m.emplace("key", "value");
```
Comment (fmanco) [inline osquery/core/tests/get_tests.cpp]: Why `emplace` as oppose to just initializing the map above?

File: osquery/core/tests/get_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  const auto& cm = m;
+  TestValue const defaultValue{"default value"};
+  {
+    auto const& exp = getOr(cm, "key", defaultValue);
+    ASSERT_EQ(exp.text, "value");
+  }
+  {
+    auto const& exp2 = getOr(cm, "no such key", defaultValue);
+    ASSERT_EQ(exp2.text, "default value");
```
Comment (fmanco) [inline osquery/core/tests/get_tests.cpp]: Any disadvantage in making `"default value"` `defaultValue`?

File: osquery/core/CMakeLists.txt [hunk 5]
```diff
@@ -15,6 +15,7 @@ target_sources(libosquery
     "${CMAKE_CURRENT_LIST_DIR}/database/database.cpp"
     "${CMAKE_CURRENT_LIST_DIR}/database/database.h"
     "${CMAKE_CURRENT_LIST_DIR}/flags.cpp"
+    "${CMAKE_CURRENT_LIST_DIR}/get.h"
```
[the remark this answers is not in the document]
Comment (akindyakov) [reply osquery/core/CMakeLists.txt]: Thank you, this is make sense. Moved to `map_take.h`, what do you think?

**Pull request merged**